include/Zoo.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "Animal.h"
+#include <vector>
+
+#define X 10
+
+enum AnimalType {
+#define ANIMAL(a) E_##a
+#include "AnimalTypes.h"
+	E_unsupported,
+	E_supported_animals_num = E_unsupported,
+#undef ANIMAL(a)
+};
+
+class Zoo {
+
+public:
+	//Zoo(std::vector<std::vector<int>> zoo_map);
+	void run();
+
+private:
+	const std::vector<std::string> knownAnimals = {
+#define ANIMAL(a) #a
```

Comment:
what is that ??

Reply:
vector for my animals to know if the user input for some animal is ok